pkg/controller/controllerexpectations.go

```diff
@@ -0,0 +1,217 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
```

Comment:
Lot of code coming in with this, I assume this was from kubernetes/kubernetes and thus not something we want to vendor due to it's size?

Reply:
Precisely. Ideally, ControllerExpectation would be moved out of kube/kube and into one of the other kube repos since it is has nothing that is specific to kube/kube.